testing/mocks/generic.go

```diff
@@ -183,31 +79,364 @@ var (
 	//        / \   \
 	//       1   2   4
 	GenericRootNode = node.NewNode(
-		42,
+		256,
 		node.NewNode(
-			42,
-			node.NewLeaf(GenericLedgerPaths[0], GenericLedgerPayloads[0], 42),
-			node.NewLeaf(GenericLedgerPaths[1], GenericLedgerPayloads[1], 42),
-			GenericLedgerPaths[2],
-			GenericLedgerPayloads[2],
+			256,
+			node.NewLeaf(GenericLedgerPath(0), GenericLedgerPayload(0), 42),
+			node.NewLeaf(GenericLedgerPath(1), GenericLedgerPayload(1), 42),
+			GenericLedgerPath(2),
+			GenericLedgerPayload(2),
 			hash.DummyHash,
 			64,
 			64,
 		),
 		node.NewNode(
-			42,
-			node.NewLeaf(GenericLedgerPaths[3], GenericLedgerPayloads[3], 42),
+			256,
+			node.NewLeaf(GenericLedgerPath(3), GenericLedgerPayload(3), 42),
 			nil,
-			GenericLedgerPaths[4],
-			GenericLedgerPayloads[4],
+			GenericLedgerPath(4),
+			GenericLedgerPayload(4),
 			hash.DummyHash,
 			64,
 			64,
 		),
-		GenericLedgerPaths[5],
-		GenericLedgerPayloads[5],
+		GenericLedgerPath(5),
+		GenericLedgerPayload(5),
 		hash.DummyHash,
 		64,
 		64,
 	)
+
+	GenericTrie, _ = trie.NewMTrie(GenericRootNode)
+
+	GenericCurrency = identifier.Currency{
+		Symbol:   dps.FlowSymbol,
+		Decimals: dps.FlowDecimals,
+	}
+	GenericAccount = flow.Account{
+		Address: GenericAddress(0),
+	}
+
+	GenericBlockQualifier = identifier.Block{
+		Index: &GenericHeight,
+		Hash:  GenericHeader.ID().String(),
+	}
 )
+
+func GenericCommits(number int) []flow.StateCommitment {
```

Comment:
I'm going to be in the minority now, but I actually preferred the previous iteration, with the static values defined, only thought the retrieval would be nicer by using functions instead of fetching directly from the array 😄 

I don't like the fact that `GenericCommit(57)` will actually burn the CPU by generating 57 commits and discard 56 of them. I think majority of the time we only do want the **one** commit/identifier/transaction/event. Plus, we have no cache so running it multiple times will just repeat all the work.

I was thinking of the option of having something like
```go
func GenericAddress(number int) flow.Address {

	seed := rand.NewSource(number)
	random := rand.New(seed)

	var address flow.Address
	binary.BigEndian.PutUint64(address[0:], random.Uint64())

	return address
}
```

but I'm not sure I like that one much either.

Reply:
I think it's an interesting idea, the only issue I have with it is it might bring a fair amount of complexity to the code of the generic functions, while I'm pretty sure it saves a few nanoseconds at best, so I'm not sure it's worth the tradeoff, at least with the way we currently use it. If at some point we do use larger generated samples (like of a size 57 like in your example) I think we could reconsider indeed, but for now I feel like it's overkill :p

Reply:
There is a pretty easy way to address this if we ever think this is a bottleneck to test times.